Form/Type/CategorySelectorType.php

```diff
@@ -50,17 +51,28 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
         $this->configureOptions($resolver);
     }
 
+    /**
+     * NEXT_MAJOR: replace usage of deprecated 'choice_list' option, when bumping requirements to SF 2.7+.
+     *
+     * {@inheritdoc}
+     */
     public function configureOptions(OptionsResolver $resolver)
     {
         $that = $this;
 
         $resolver->setDefaults(array(
             'context' => null,
             'category' => null,
-            'choice_list' => function (Options $opts, $previousValue) use ($that) {
-                return new SimpleChoiceList($that->getChoices($opts));
-            },
         ));
+        if (class_exists('Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface')) {
+            $resolver->setDefault('choice_loader', function (Options $opts, $previousValue) use ($that) {
+                return new CategoryChoiceLoader(array_flip($that->getChoices($opts)));
+            });
+        } else {
```

Comment:
Please invert the condition and use an early return statement (so no else), so that the next major things can be done just by removing an if block.